### PR TITLE
Device details properties & styling

### PR DESCRIPTION
**Pull request by mechaffin**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

- Remove redundant properties (supportedMethods, telemetry) from display.
- Restyle grids (tags, methods, and properties) per xD specs.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 1]
```diff
@@ -116,85 +115,111 @@ export class DeviceDetails extends Component {
...
-                      <HeaderCell className="col-7">{t('devices.details.properties.valueHeader')}</HeaderCell>
-                    </Row>
-                    {
-                      properties.map(([propertyName, propertyValue], idx) =>
-                        <Row key={idx}>
-                          <Cell className="col-3">{propertyName}</Cell>
-                          <Cell className="col-7">{propertyValue.toString().replace(/,/g, ", ")}</Cell>
+                      </GridHeader>
+                      <GridBody>
+
```
Comment (ppathan) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: nit: extra line

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 2]
```diff
@@ -116,85 +121,122 @@ export class DeviceDetails extends Component {
...
-                  </Grid>
-                }
-              </Accordion>
-
-              <Accordion title={t('devices.details.diagnostics.title')} description={t('devices.details.diagnostics.description')}>
-                TODO: Add diagnostics.
-              </Accordion>
+                      </GridHeader>
+                      <GridBody>
+
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: Remove the new line

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 3]
```diff
@@ -116,85 +121,122 @@ export class DeviceDetails extends Component {
...
+                <Section.Header>{t('devices.details.properties.title')}</Section.Header>
+                <Section.Content>
+                  <SectionDesc>{t('devices.details.properties.description')}</SectionDesc>
+                  {
+                    (properties.length === 0) &&
+                    t('devices.details.properties.noneExist')
+                  }
+                  {
+                    (properties.length > 0) && [
+                      <Grid>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: I think you need a `key` here. Isn't there a warning in the console?

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 4]
```diff
@@ -116,85 +121,122 @@ export class DeviceDetails extends Component {
...
+                            properties.map(([propertyName, propertyValue], idx) =>
+                              <Row key={idx}>
+                                <Cell className="col-3">{propertyName}</Cell>
+                                <Cell className="col-7">{propertyValue.toString().replace(/,/g, ", ")}</Cell>
+                              </Row>
+                            )
+                          }
+                        </GridBody>
+                      </Grid>,
+                      <BtnToolbar className="copy-device-properties">
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: Doesn't seem like BtnToolbar is needed. All it does add a flex box.

File: src/components/shared/flyout/flyoutSection/flyoutSection.scss [hunk 5]
```diff
@@ -39,7 +39,7 @@
   }
 
   @include themify($themes) {
-    border-top: 1px solid themed('colorMidSecond'); // TODO: Move to flyout scoped colors
+    border-top: 1px solid themed('colorFlyoutAccordionBorder');
```
Comment (spryor) [inline src/components/shared/flyout/flyoutSection/flyoutSection.scss]: `colorFlyoutSectionBorder`

File: src/components/pages/devices/flyouts/deviceDetails/propertyGrid/propertyGridBody.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/propertyGrid/propertyGridBody.js]: You don't need to do this in the JS. You can actually do it all in the CSS. Take a look at [this example](https://jsfiddle.net/1n0axa3y/110/) I put together. In this example, I can add or remove the div with the `scrollable` class name and the grid will convert from a scrollable to a non-scrollable grid automatically with just CSS and minimal markup changes. It is also variable easy to read the JSX and understand what each component is doing.

**Pull request merged**